dependency-injection/DependencyInjection/Controller.cs

```diff
@@ -3,12 +3,28 @@
 namespace DependencyInjection;
 
 [ApiController]
-public class Controller(Singleton _singleton)
+public class Controller(IServiceProvider _serviceProvider, Singleton _singleton)
 {
     [HttpPost]
     [Route("stuff")]
     public void DoStuff()
     {
         _singleton.DoStuff("controller");
     }
+
+    [HttpPost]
+    [Route("from-keyed-services/personal-a")]
+    public string FromKeyedServicesUsing([FromKeyedServices(ServiceImplementation.PersonalA)] IPersonal personal)
+    {
+        return personal.Name;
+    }
```

Comment:
conform to naming conventions even for samples like these. `FromKeyedServicesUsing` is not a proper method name